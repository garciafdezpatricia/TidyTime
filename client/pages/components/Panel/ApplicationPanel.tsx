import { useState } from "react";
import Toggle from "../ToggleSwitch/ToggleSwitch";
import { Icon } from "../Icon/Icon";


export default function ApplicationPanel() {

    const [change, setChange] = useState(false);


    return (
        <article className="applications-panel">
            <h3>Connect to applications</h3>
            <hr></hr>
            <div className="panel-content">
                <Application 
                    appImg={"./google.svg"} 
                    appAlt={"Google icon"} 
                    appName={"Google Calendar"} 
                    appDesc={"Connect to Google to manage your events, calendars and tasks."}
                />
                <Application 
                    appImg={"./github.svg"} 
                    appAlt={"GitHub icon"} 
                    appName={"GitHub"} 
                    appDesc={"Connect to GitHub to manage your issues."}
                />
            </div>
        </article>
    )
}

function Application({appImg, appAlt, appName, appDesc}:
    {
        appImg:string, 
        appAlt:string,
        appName:string,
        appDesc:string,
    }) {
    return (
        <section className="content-box">
            <section className="content-header">
                <div className="header-app">
                    <Icon className="app-icon" src={appImg} alt={appAlt} />
                    <h4>{appName}</h4>
                </div>
                <Toggle isChecked={false} onChange={() => {}}/>
            </section>
            <section className="content-body">
                <p>{appDesc}</p>
            </section>
        </section>
    )
}