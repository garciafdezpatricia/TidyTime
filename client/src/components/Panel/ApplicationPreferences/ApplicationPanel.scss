@use "../../../../styles/variables" as *;

.applications-panel {
    padding: 1rem;
    display: flex;
    flex-direction: column;
    width: calc(100% - 2rem);
}

.panel-content {
    padding-top: 1.5rem;
    display: flex;
    gap: 2rem;
    .content-box {
        border: .1rem solid $light-grey;
        border-radius: .4rem;
        display: grid;
        grid-template-rows: .5fr 1fr;
        grid-template-columns: 1fr 1fr;
        width: 18rem;

        .content-header {
            padding: .5rem 1rem;
            border-top-left-radius: .4rem;
            border-top-right-radius: .4rem;
            grid-row: 1/1;
            grid-column: 1/-1;
            border-bottom: .1rem solid $light-grey;
            display: flex;
            align-items: center;
            background-color: $primary-grey;
            justify-content: space-between;

            .header-app {
                display: flex;
                gap: .6rem;
            }

            .app-icon {
                width: 2rem;
                height: auto;
            }
        }

        .content-body {
            font-size: .8rem;
            padding: .5rem;
            grid-row: 2/2;
            grid-column: 1/-1;
        }
    }
}