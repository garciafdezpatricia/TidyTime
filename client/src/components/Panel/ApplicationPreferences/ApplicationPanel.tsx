import Toggle from "../../ToggleSwitch/ToggleSwitch";
import { Icon } from "../../Icon/Icon";
import { useGoogleContext } from "../../Context/GoogleContext";
import { useGoogleHandler } from "@/pages/api/google";
import { useGithubContext } from "../../Context/GithubContext";
import { useGithubHandler } from "@/pages/api/github";
import { useEffect, useState } from "react";
import { useTranslation } from 'react-i18next';


export default function ApplicationPanel() {
    const { t } = useTranslation();
    const { loggedIn, authUrl } = useGoogleContext();
    const { githubLoggedIn } = useGithubContext();
    const { handleLogout, handleLogin } = useGoogleHandler();
    const { logoutGithub, loginWithGithub } = useGithubHandler();

    const [reRender, setRerender] = useState(Math.random());

    useEffect(() => {
        handleLogin();
    }, [reRender])
    
    const googleSwitch = (value:boolean) => {
        if (value) {
            console.log(authUrl);
            window.location.assign(authUrl);
        } else {
            handleLogout();
        }
    }

    const githubSwitch = async (value:boolean) => {
        if (value) {
            await loginWithGithub();
        } else {
            await logoutGithub();
        }
    }

    return (
        <article className="applications-panel">
            <h3>{t('preferences.applications.title')}</h3>
            <hr></hr>
            <p>{t('preferences.applications.desc')}</p>
            <div className="panel-content">
                <Application 
                    appImg={"./google.svg"} 
                    appAlt={"Google icon"} 
                    appName={"Google Calendar"} 
                    appDesc={t('preferences.applications.google')}
                    checked={loggedIn}
                    onChange={googleSwitch}
                />
                <Application 
                    appImg={"./github.svg"} 
                    appAlt={"GitHub icon"} 
                    appName={"GitHub"} 
                    appDesc={t('preferences.applications.github')}
                    checked={githubLoggedIn}
                    onChange={githubSwitch}
                />
            </div>
        </article>
    )
}

export function Application({appImg, appAlt, appName, appDesc, checked, onChange}:
    {
        appImg:string, 
        appAlt:string,
        appName:string,
        appDesc:string,
        checked?: boolean,
        onChange: (arg?:any) => void | any;
    }) {
    return (
        <section className="content-box">
            <section className="content-header">
                <div className="header-app">
                    <Icon className="app-icon" src={appImg} alt={appAlt} />
                    <h4>{appName}</h4>
                </div>
                <Toggle isChecked={checked ?? false} onChange={(value) => onChange(value)}/>
            </section>
            <section className="content-body">
                <p>{appDesc}</p>
            </section>
        </section>
    )
}