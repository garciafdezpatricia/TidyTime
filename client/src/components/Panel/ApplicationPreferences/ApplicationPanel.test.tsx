import { render } from "@testing-library/react";
import '@testing-library/jest-dom';
import {Application} from '@/src/components/Panel/ApplicationPreferences/ApplicationPanel';

describe('Application', () => {
    it('renders without crashing', () => {
      const { container } = render(<Application appImg={""} appAlt={""} appName={""} appDesc={""} onChange={function (arg?: any) {
        throw new Error("Function not implemented.");
      } } />);
      expect(container).toBeInTheDocument();
    });

    it('matches snapshot', () => {
        const { asFragment } = render(<Application appImg={""} appAlt={""} appName={""} appDesc={""} onChange={function (arg?: any) {
          throw new Error("Function not implemented.");
        } } />);
        expect(asFragment()).toMatchSnapshot();
      });
  });