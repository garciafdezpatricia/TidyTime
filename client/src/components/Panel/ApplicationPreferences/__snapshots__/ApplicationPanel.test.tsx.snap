// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Application matches snapshot 1`] = `
<DocumentFragment>
  <section
    class="content-box"
  >
    <section
      class="content-header"
    >
      <div
        class="header-app"
      >
        <img
          alt=""
          class="app-icon"
          src=""
        />
        <h4 />
      </div>
      <label
        class="switch"
      >
        <input
          type="checkbox"
        />
        <span
          class="slider round"
        />
      </label>
    </section>
    <section
      class="content-body"
    >
      <p />
    </section>
  </section>
</DocumentFragment>
`;
